torch/nn/modules/utils.py

```diff
@@ -13,3 +13,11 @@ def parse(x):
 _pair = _ntuple(2)
 _triple = _ntuple(3)
 _quadruple = _ntuple(4)
+
+
+def _list_with_default(out_size, defaults):
+    if isinstance(out_size, int):
+        return out_size
+    if len(defaults) <= len(out_size):
+        raise ValueError('Input dimension should be at least {}'.format(len(out_size) + 1))
```

Comment:
nit: this will be confusing as the adaptive methods will actually fail if input dim is no at least **2** + output_size. Can you try to make the error message nicer?

Reply:
So this is another discrepancy between our C backend(3D input actually also works with MaxPool2d) although it's not in our documentation. I agree that we should fix it based on doc so that it's easier to the TH->ATen migration as well.

Reply:
I see. Thanks for the explanation!